keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -353,6 +363,38 @@ Contains a unique identifier for the containing cluster.
 
 **Reusing cluster names**: Since an `id.k8s.io ClusterProperty` has no restrictions on whether or not a ClusterProperty can be repeatable, if a cluster unregisters from a ClusterSet it is permitted under this standard to rejoin later with the same `id.k8s.io ClusterProperty` it had before. Similarly, a *different* cluster could join a ClusterSet with the same `id.k8s.io ClusterProperty` that had been used by another cluster previously, as long as both do not have membership in the same ClusterSet at the same time. Finally, two or more clusters may have the same `id.k8s.io ClusterProperty` concurrently (though they **should** not; see "Uniqueness" above) *as long as* they both do not have membership in the same ClusterSet.
 
+#### Property: `network.k8s.io`
+
+Contains an identifier representing the network for the cluster.
+
+
+##### Uniqueness
+
+*   The identifier **need not be** unique but **should** remain the same for a cluster for the duration of the cluster’s membership.
+
+##### Lifespan
```

Comment:
We should think about under what circumstances this should (or must) exist. If network.k8s.io is filled out in one cluster in the ClusterSet, is it filled out in all of them? At what time in a cluster's connectivity to the rest of the ClsuterSet is this expected to exist? If connectivity for only one cluster is removed (or lost?), can/should its network.k8s.io value be removed (and does that have any bearing on what a consumer expects regarding this being filled out across the ClusterSet?)

Reply:
The property is expected to exist when cluster joins the ClusterSet and remain the same for rest of the clusterset membership duration. An MCS controller would use this property to generate the `EndpointSlice`s for the cluster. An MCS controller will read and cache this property for a cluster and use this information to generate the EndpointSlice for a cluster in imported clusters based on their network.k8s.io value. If the network changes or goes missing after its first read then the EndpointSlice may be updated in the clusters accordingly (goes missing - default network or flat network, modified - a updated network name which might or might not already exist on some other cluster). Hence, `should` makes sense for the immutability of `network.k8s.io`